### PR TITLE
Fix view for orders api (Fixes Issue #2512)

### PR DESCRIPTION
**Pull request by skukx**
Refactor _big.json.jbuilder so that it dynamically loads the payment
source view based on that source's payment method.

Add payment views for each payment method which will be used to render a
payment source as json.

Fixes https://github.com/solidusio/solidus/issues/2512.

### CODE REVIEW

**Comment by skukx** [comment]
Yeah I can add a test for this. As far as the naming convention, I followed convention on how Spree admin chooses to display payment sources.

See: https://github.com/solidusio/solidus/tree/master/backend/app/views/spree/admin/payments/source_views

If we want to move away from that convention I will change the views mimic that payment source rather than the method

**Comment by mamhoff** [comment]
Sorry, I thought we changed the partial name with deprecating the `Spree::Gateway` class, but we didn't: 
https://github.com/solidusio/solidus/blob/master/core/app/models/spree/payment_method/credit_card.rb#L15

So, keep the partial name, but do add a spec. :heart:

**Comment by skukx** [comment]
Let me know if the specs added are sufficient

File: api/spec/requests/spree/api/orders_controller_spec.rb [hunk 1]
```diff
@@ -317,6 +317,21 @@ module Spree
           expect(credit_cards[0]['id']).to eq payment.source.id
           expect(credit_cards[0]['address']['id']).to eq credit_card.address_id
         end
+
+        it 'renders the payment source view for gateway' do
+          subject
+          expect(response).to render_template partial: "spree/api/payments/source_views/_#{payment.payment_method.partial_name}"
```
Comment (mamhoff) [inline api/spec/requests/spree/api/orders_controller_spec.rb]: These are sufficient, but should be more explicit. Either `render_template('spree/api/payments/source_views/_gateway.html.erb`), or test directly for the right thing being rendered as JSON.

**Comment by skukx** [comment]
Made changes as requested

File: api/app/views/spree/api/orders/_big.json.jbuilder [hunk 2]
```diff
@@ -25,11 +25,10 @@ json.payments(order.payments) do |payment|
   json.payment_method { json.(payment.payment_method, :id, :name) }
   json.source do
     if payment.source
```
Comment (skukx) [inline api/app/views/spree/api/orders/_big.json.jbuilder]: It is possible that `payment.source` is a `Spree::Payment`.  
See: https://github.com/solidusio/solidus/blob/v2.4/backend/app/views/spree/admin/payments/show.html.erb#L16

We will need to account for that situation here as well:
```ruby
json.source do
    ##
    # payment.source could be a Spree::Payment. If it is then we need to call
    # source twice.
    # @see https://github.com/solidusio/solidus/blob/v2.4/backend/app/views/spree/admin/payments/show.html.erb#L16
    #
    payment_source = payment.source.is_a?(Spree::Payment) ? payment.source.source : payment.source
    
    if payment_source
      json.partial!(
        "spree/api/payments/source_views/#{payment.payment_method.partial_name}",
        payment_source: payment_source
      )
    else
      json.nil!
    end
  end
```

When I get a chance I'll add this to the PR

**Comment by skukx** [comment]
What is the status of this?

**Comment by gmacdougall** [comment]
There's a red CircleCI build on this right now. Do you know the source of the errors?

**Comment by skukx** [comment]
I just pushed something new so it may have to do with that. I'll take a look

**Comment by skukx** [comment]
Look like I just needed rebase my branch on master

**Pull request merged**